### PR TITLE
Drop the 'file' store loader

### PR DESCRIPTION
**Pull request by levitte**
The TPM2 'file' store loader isn't necessary, so we remove it.

The idea is that the OpenSSL 'file' store loader should be agnostic
enough to simply rely on available decoders and adjoined keymgmt
implementations.  This includes the TPM2 decoders, when that provider
is loaded.

Signed-off-by: Richard Levitte <richard@levitte.org>

### CODE REVIEW

File: src/tpm2-provider.c [hunk 1]
```diff
@@ -234,7 +234,6 @@ extern const OSSL_DISPATCH tpm2_file_store_functions[];
 extern const OSSL_DISPATCH tpm2_object_store_functions[];
 
 static const OSSL_ALGORITHM tpm2_stores[] = {
-    { "file", TPM2_PROPS(store), tpm2_file_store_functions },
```
Comment (gotthardp) [inline src/tpm2-provider.c:237]: Could you please remove also the `extern const OSSL_DISPATCH tpm2_file_store_functions[]` on line 233?

File: test/ecdsa_genpkey_auth.sh [hunk 2]
```diff
@@ -6,15 +6,15 @@ set -eufx
 echo -n "abcde12345abcde12345abcde12345ab" > testdata
 
 # generate private key as PEM
-openssl genpkey -provider tpm2 -algorithm EC -pkeyopt group:P-256 \
+openssl genpkey -provider tpm2 -provider base -algorithm EC -pkeyopt group:P-256 \
```
Comment (gotthardp) [inline test/ecdsa_genpkey_auth.sh:9]: Is the `-provider base` really necessary here? There are no file inputs.

File: src/tpm2-provider.c (hunk 1 above)
Comment (levitte) [reply src/tpm2-provider.c:237]: Absolutely!

File: test/ecdsa_genpkey_auth.sh (hunk 2 above)
Comment (levitte) [reply test/ecdsa_genpkey_auth.sh]: Uhm...  apparently not.  Removed

File: test/ecdsa_genpkey_sign_rawin.sh [hunk 3]
```diff
@@ -5,18 +5,18 @@ set -eufx
 echo -n "abcde12345abcde12345" > testdata
 
 # generate private key as PEM
-openssl genpkey -provider tpm2 -algorithm EC -pkeyopt group:P-256 -out testkey.priv
+openssl genpkey -provider tpm2 -provider base -algorithm EC -pkeyopt group:P-256 -out testkey.priv
```
Comment (gotthardp) [inline test/ecdsa_genpkey_sign_rawin.sh:8]: I am sorry, my previous comment was incomplete: this `-provider base` feels superfluous too.

File: test/rsa_genpkey_auth_parent.sh [hunk 4]
```diff
@@ -11,15 +11,15 @@ tpm2_createprimary -G rsa -g sha256 -c parent.ctx
 HANDLE=$(tpm2_evictcontrol -c parent.ctx | cut -d ' ' -f 2 | head -n 1)
 
 # generate key with an user authorization
-openssl genpkey -provider tpm2 -algorithm RSA -out testkey.priv \
+openssl genpkey -provider tpm2 -provider base -algorithm RSA -out testkey.priv \
```
Comment (gotthardp) [inline test/rsa_genpkey_auth_parent.sh:14]: And this.

File: test/rsa_genpkey_sign.sh [hunk 5]
```diff
@@ -6,15 +6,15 @@ set -eufx
 echo -n "abcde12345abcde12345abcde12345ab" > testdata
 
 # generate key with no scheme/hash constraints
-openssl genpkey -provider tpm2 -algorithm RSA -pkeyopt bits:1024 -out testkey.priv
+openssl genpkey -provider tpm2 -provider base -algorithm RSA -pkeyopt bits:1024 -out testkey.priv
```
Comment (gotthardp) [inline test/rsa_genpkey_sign.sh:9]: And this.

File: test/ecdsa_genpkey_sign_rawin.sh (hunk 3 above)
Comment (levitte) [reply test/ecdsa_genpkey_sign_rawin.sh:8]: Nothing to be sorry about.  I'm grateful for your attention for details! 😃 

Fixed

File: test/rsa_genpkey_sign.sh (hunk 5 above)
Comment (levitte) [reply test/rsa_genpkey_sign.sh]: ... and done

**Pull request merged**